packages/astro/src/core/build/generate.ts

```diff
@@ -196,9 +196,12 @@ export async function generatePages(opts: StaticBuildOptions, internals: BuildIn
 		}
 	}
 
-	logger.info(null, `\n${bgGreen(black(` generating optimized images `))}`);
-	for (const imageData of getStaticImageList()) {
-		await generateImage(pipeline, imageData[1].options, imageData[1].path);
+	const staticImageList = Array.from(getStaticImageList())
```

Comment:
Oh, ok. I didn't want to actually change the return value, to keep the changes to a minimum.

Will do!

Reply:
Yeah, no worries here since it's only called here. The biggest reason it's in a function is to "sandbox" the usage of the `globalThis.astroAsset` because we're hoping to change the implementation eventually